CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_StaffSlots_Infiltration.uc

```diff
@@ -127,3 +133,83 @@ static function string GetResistanceRingBonusDisplayString(XComGameState_StaffSl
+{
+	local X2StaffSlotTemplate Template;
+
+	Template = CreateStaffSlotTemplate('InfiltrationStaffSlot');
+
+	// Same as default slot
+	Template.bPreventFilledPopup = true;
+	Template.bSoldierSlot = true;
+	Template.FillFn = class'X2StrategyElement_XpackStaffSlots'.static.FillCovertActionSlot;
+	Template.CanStaffBeMovedFn = class'X2StrategyElement_XpackStaffSlots'.static.CanStaffBeMovedCovertActions;
+	Template.GetNameDisplayStringFn = class'X2StrategyElement_XpackStaffSlots'.static.GetCovertActionSoldierNameDisplayString;
+	Template.IsUnitValidForSlotFn = class'X2StrategyElement_XpackStaffSlots'.static.IsUnitValidForCovertActionSoldierSlot;
+	
+	// Custom
+	Template.EmptyFn = EmptyInfiltrationSlot;
+	Template.GetLocationDisplayStringFn = GetInfiltrationLocationString;
+
+	return Template;
+}
+
+static function EmptyInfiltrationSlot(XComGameState NewGameState, StateObjectReference SlotRef)
+{
+	local XComGameState_Unit NewUnitState;
+	local XComGameState_StaffSlot NewSlotState;
+	local XComGameState_CovertAction NewActionState;
+
+	EmptySlot(NewGameState, SlotRef, NewSlotState, NewUnitState);
+
+	if (NewSlotState.GetCovertAction().bCompleted)
+	{
+		// This is an inflitration that is ready to go
+		NewUnitState.SetStatus(eStatus_Active);
+		return;
+	}
+	
+	// Otherwise ee are still in loadout, so do the default things
+	// The code below is copy-paste from X2StrategyElement_XpackStaffSlots::EmptyCovertActionSlot
+
+	// Only set a unit status back to normal if they are still listed as being on the covert action
+	// Since this means they weren't killed, wounded, or captured on the mission
+	if (NewUnitState.GetStatus() == eStatus_CovertAction)
+	{
+		NewUnitState.SetStatus(eStatus_Active);
+
+		// Don't change super soldier loadouts since they have specialized gear
+		if (NewUnitState.IsSoldier() && !NewUnitState.bIsSuperSoldier)
+		{
+			// First try to upgrade the soldier's primary weapons, in case a tier upgrade happened while
+			// they were away on the CA. This is needed to make sure weapon upgrades and customization transfer properly.
+			// Issue #230 start
+			//CheckToUpgradePrimaryWeapons(NewGameState, NewUnitState);
+			class'X2StrategyElement_XpackStaffSlots'.static.CheckToUpgradeItems(NewGameState, NewUnitState);
+			// Issue #230 end
+
+			// Then try to equip the rest of the old items
+			if (!class'CHHelpers'.default.bDontUnequipCovertOps) // Issue #153
+			{
+				NewUnitState.EquipOldItems(NewGameState);
+			}
+
+			// Try to restart any psi training projects
+			class'XComGameStateContext_StrategyGameRule'.static.PostMissionUpdatePsiOperativeTraining(NewGameState, NewUnitState);
+		}
+	}
+
+	NewActionState = GetNewCovertActionState(NewGameState, NewSlotState);
+	NewActionState.UpdateNegatedRisks(NewGameState);
+	NewActionState.UpdateDurationForBondmates(NewGameState);
+}
+
+static function string GetInfiltrationLocationString(XComGameState_StaffSlot SlotState)
+{
+	return "Infiltration";
```

Comment:
Localize!

Reply:
This doesn't even show up in the UI for some reason 😂

Reply:
I think it's supposed to show in the barracks soldier list as the unit status?

Reply:
i thought so as well, but it's just empty space there....